### PR TITLE
WIP - Adding QM/MM calculator

### PR DESCRIPTION
**Pull request by jgolebiowski**
Adding the calculator for Hybrid Atomistic Simulations of Entangled Systems

### CODE REVIEW

**Comment by jameskermode** [comment]
Thanks @jgolebiowski  - there are a few style issues to be resolved before this gets merged, as discussed.

**Comment by jameskermode** [comment]
@pastewka Failing CI tests are on Mac OS only - did you ever get to the bottom of why this happens? Seems to be some configuration problem.

File: matscipy/calculators/mcfm/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+###############################################################
+###     INIT file for the hades functions module
```
Comment (jameskermode) [inline matscipy/calculators/mcfm/__init__.py]: should hades be changed to mcfm everywhere?

File: matscipy/calculators/mcfm/neighbour_list_mcfm/neighbour_list_base.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import numpy as np
+
+
+class NeighbourListBase(object):
+    """Interface for the neighbour list.
+    HADES module can use any neighbour list object as long
```
Comment (jameskermode) [inline matscipy/calculators/mcfm/neighbour_list_mcfm/neighbour_list_base.py]: HADES -> MCFM

File: matscipy/calculators/mcfm/qm_cluster.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import, division, print_function, unicode_literals
+import numpy as np
+
+from .qm_cluster_tools.qm_flagging_tool import QMFlaggingTool
+from .qm_cluster_tools.qm_clustering_tool import QMClusteringTool
+
+
+class QMcluster(object):
```
Comment (jameskermode) [inline matscipy/calculators/mcfm/qm_cluster.py]: maybe capitalisation `QMCluster` is better? but not a big deal.

File: matscipy/calculators/mcfm/qm_cluster.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    It acts as a mediator between
+    -----------------------------
+        QM flagginig module
+        QM clustering module
+        neighbour list
+    and contains interface to those classes
+
+    Attributes
+    ----------
+    clustering_module : hades.qmClusterModule.QMClusteringTool
```
Comment (jameskermode) [inline matscipy/calculators/mcfm/qm_cluster.py]: hades -> matscipy.calculators.mcfm

**Comment by pastewka** [comment]
Not sure why the Mac tests fail. The current error is that it cannot find the compiler, which is odd. I had disable clang at some point because that did not seem to work.

**Comment by jgolebiowski** [comment]
I have now updated the documentation to be more consistent (hades -> mcfm). I completely forgot about it before.

**Comment by jameskermode** [comment]
Thanks Jacek.

@pastewka could you have another quick look and see if you are now happy for this to be merged?

**Pull request merged**